code/src/java/pcgen/core/RollingMethods.java

```diff
@@ -216,7 +216,7 @@ private static int roll(
         return IntStream.generate(() -> roll(sides - reroll) + reroll)
             .limit(times)
             .sorted()
-            .limit(numToKeep)
+            .skip(times-numToKeep)
```

Comment:
Another option is to keep this line and replaced .sorted() with `.sorted(Comparator.reverseOrder())`